src/prefect/core/flow.py

```diff
@@ -1463,6 +1465,21 @@ def serialize(self, build: bool = False) -> dict:
 
         return serialized
 
+    def serialized_hash(self, build: bool = False) -> str:
+        """
+        Generate a deterministic hash of the serialized flow. This is useful for
+        determining if the flow has changed. If this hash is equal to a previous hash,
+        no new information would be passed to the server on a call to `flow.register()`
+
+        Args:
+            - build (bool, optional):  if `True`, the flow's environment is built
+                prior to serialization. Passed through to `Flow.serialize()`.
+
+        Returns:
+            - str: the hash of the serialized flow
+        """
+        return hashlib.sha256(json.dumps(self.serialize(build)).encode()).hexdigest()
```

Comment:
@madkinsz apparently this is not determenistic, because `self.serialize` may return OrderedDict with common dicts in its values, which can have random order upon json serialization

Reply:
A quick fix here could be adding `sort_keys=True` to json.dumps

Reply:
Thanks for the note! This is fixed in #3654 and should be in the next release.